src/loaders/JsonImageInfoLoader.ts

```diff
@@ -187,57 +198,90 @@ class JsonImageInfoLoader implements IVolumeLoader {
       const url = imageArray[i].name;
       const batch = imageArray[i].channels;
 
-      // using Image is just a trick to download the bits as a png.
-      // the Image will never be used again.
-      const img: HTMLImageElement = new Image();
-      img.onerror = function () {
-        console.log("ERROR LOADING " + url);
+      const regionPx = convertSubregionToPixels(volume.loadSpec.subregion, volume.imageInfo.subregionSize);
+      const cacheQueryDims = {
+        region: regionPx,
+        time: volume.loadSpec.time,
+        scale: 0,
       };
-      img.onload = (function (thisbatch) {
-        return function (event: Event) {
-          //console.log("GOT ch " + me.src);
-          // extract pixels by drawing to canvas
-          const canvas = document.createElement("canvas");
-          // nice thing about this is i could downsample here
-          const w = Math.floor((event?.target as HTMLImageElement).naturalWidth);
-          const h = Math.floor((event?.target as HTMLImageElement).naturalHeight);
-          canvas.setAttribute("width", "" + w);
-          canvas.setAttribute("height", "" + h);
-          const ctx = canvas.getContext("2d");
-          if (!ctx) {
-            console.log("Error creating canvas 2d context for " + url);
-            return;
-          }
-          ctx.globalCompositeOperation = "copy";
-          ctx.globalAlpha = 1.0;
-          ctx.drawImage(event?.target as CanvasImageSource, 0, 0, w, h);
-          // getImageData returns rgba.
-          // optimize: collapse rgba to single channel arrays
-          const iData = ctx.getImageData(0, 0, w, h);
-
-          const channelsBits: Uint8Array[] = [];
-          // allocate channels in batch
-          for (let ch = 0; ch < Math.min(thisbatch.length, 4); ++ch) {
-            channelsBits.push(new Uint8Array(w * h));
-          }
-          // extract the data
-          for (let j = 0; j < Math.min(thisbatch.length, 4); ++j) {
-            for (let px = 0; px < w * h; px++) {
-              channelsBits[j][px] = iData.data[px * 4 + j];
+      // if any in batch is cached then they all are...?
```

Comment:
What's the uncertainty here? Is it possible for only part of a batch to be cached?

Reply:
I think I might have a go at rewriting this after Kevin's comments about it.  It could be a bit simpler.  The question mark is just me knowing that I am baking in a very specific assumption and the logic only works because of it.